sqlair.go

```diff
@@ -476,14 +422,14 @@ func (tx *TX) Query(ctx context.Context, s *Statement, inputArgs ...any) *Query
 		return &Query{ctx: ctx, err: err}
 	}
 
-	sqlstmt, ok := stmtCache.lookupStmt(tx.db, s)
-	if !ok {
-		sqlstmt, err = tx.sqlconn.PrepareContext(ctx, pq.SQL())
-		if err != nil {
-			return &Query{ctx: ctx, err: err}
+	run := func() (rows *sql.Rows, result sql.Result, err error) {
```

Comment:
The amount of knowledge you need from `pq` in both places where this closure is generated suggests that it might be better sourced from `PrimedQuery`.

Reply:
You mean having a function in `PrimedQuery` that takes a TX or a DB and returns the closure?

I put it in `sqlair.go` because one of the project organization ideas was to not use the `sql` package inside the internal packages (there are only some exceptions right now such as when creating the named parameters). Do you think it is better to move it?

Reply:
Yes, that's what I mean. I'd move it, particularly if there are already departures from the convention.

Reply:
I have had a go at implementing the change and we would need to revive the `PrepareSubstrate` interface which we just removed in other to be able to pass `TX` or `DB` to the same function (that will return the closure). Additionally, when implementing the cache we are going to want to move the function back out again.

For context, the only exceptions today are:

bindinputs.go:63
```go
namedInput := sql.Named("sqlair_"+strconv.Itoa(inputCount), val.Interface())
```
member.go:10
```go
var scannerInterface = reflect.TypeOf((*sql.Scanner)(nil)).Elem()
```